netplan_cli/cli/commands/status.py

```diff
@@ -88,18 +88,164 @@ def pretty_print(self, data: JSON, total: int, _console_width=None) -> None:
+        return []
+
+    def _get_missing_netplan_nameservers(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_nameservers_addresses', [])
+        return []
+
+    def _get_missing_netplan_search(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_nameservers_search', [])
+        return []
+
+    def _get_missing_system_search(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_nameservers_search', [])
+        return []
+
+    def _get_missing_system_macaddress(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_macaddress')
+        return None
+
+    def _get_missing_netplan_routes(self, ifname) -> set[NetplanRoute]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_routes', set())
+        return set()
+
+    def _get_missing_system_routes(self, ifname) -> set[NetplanRoute]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_routes', set())
+        return set()
+
+    def _is_missing_dhcp4_address(self, ifname) -> bool:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_dhcp4_address', False)
+        return False
+
+    def _is_missing_dhcp6_address(self, ifname) -> bool:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_dhcp6_address', False)
+        return False
+
+    def _get_missing_system_bond_link(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_bond_link', '')
+        return ''
+
+    def _get_missing_netplan_bond_link(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_bond_link', '')
+        return ''
+
+    def _get_missing_system_bridge_link(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_bridge_link', '')
+        return ''
```

Comment:
question: In `_get_missing_system_macaddress` we're returning `None` for an undefined string, here (and in some other, similar methods) it's the empty string.. Why the difference?

Reply:
No reason, I don't why I did that :)